packages/angular_devkit/build_optimizer/src/transforms/import-tslib.ts

```diff
@@ -34,11 +34,13 @@ export function getImportTslibTransformer(): ts.TransformerFactory<ts.SourceFile
           const declarations = node.declarationList.declarations;
 
           if (declarations.length === 1 && ts.isIdentifier(declarations[0].name)) {
-            const name = (declarations[0].name as ts.Identifier).text;
+            const name = declarations[0].name.text;
+            const helperName = name.split('$')[0];
```

Comment:
I think some additional checks here would be good to prevent false positives.
If the split output length is greater than 2 or the second part is not a number, it's probably something else.

Reply:
I changed this to a RegExp

Reply:
I tend to avoid regex in performance critical areas but this transformer in general is in need of a performance refactor so it's probably not a problem.